galaxy_ng/tests/integration/api/test_sync_stage.py

```diff
@@ -0,0 +1,78 @@
+import logging
+
+import pytest
+from galaxykit.collections import upload_artifact, delete_collection
+
+from ..conftest import get_galaxy_client, get_ansible_config_sync
+from ..utils import wait_for_task, get_client, set_certification
+from orionutils.generator import build_collection
+from ..utils.iqe_utils import is_sync_testing, get_all_collections, retrieve_collection
+from ..utils.tools import generate_random_artifact_version, uuid4
+
+pytestmark = pytest.mark.qa  # noqa: F821
+
+logger = logging.getLogger(__name__)
+
+
+def start_sync(api_client, repo):
+    logger.debug(f"Syncing {repo} repo")
+    url = f'content/{repo}/v3/sync/'
+    resp = api_client(url, method="POST")
+    resp = wait_for_task(api_client, resp["task"], raise_on_error=True)
+    logger.debug(f"Response from wait_for_task_id {resp}!")
+
+
+@pytest.mark.sync
+@pytest.mark.skipif(not is_sync_testing(), reason="This test can only be run on sync-tests mode")
+def test_sync():
+    config_sync = get_ansible_config_sync()
+    galaxy_client = get_galaxy_client(config_sync)
+    gc_remote = galaxy_client("remote_admin", remote=True)
+    user_stage = gc_remote.username
+    pass_stage = gc_remote.password
+    # upload artifact to stage
+    test_version = generate_random_artifact_version()
+    namespace_name = f"namespace_{uuid4()}"
+    namespace_name = namespace_name.replace("-", "")
+    gc_remote.create_namespace(namespace_name, "system:partner-engineers")
+
+    artifact = build_collection(
+        "skeleton",
+        config={"namespace": namespace_name, "version": test_version},
+    )
+    logger.debug(f"Uploading artifact name {artifact.name} version {artifact.version}")
+    resp = upload_artifact(None, gc_remote, artifact)
+    api_client_remote = get_client(
+        {"url": "https://console.stage.redhat.com/api/automation-hub/",
+         "username": user_stage, "password": pass_stage,
+         "use_move_endpoint": True, "upload_signatures": True},
+        request_token=True, require_auth=True)
```

Comment:
what is the reason you need a `gc_remote` client and an `api_client_remote`? ... in other tests we just use one client

https://github.com/ansible/galaxy_ng/blob/2a5963fb658949a5f9ca8d9767a51e5e415b13f2/galaxy_ng/tests/integration/api/test_collection_signing.py#L68-L69

Reply:
because we should transition to using galaxykit (gc_remote) but there're methods or calls to the api that are still not available there so that's why I still need to use api_client_remote

Reply:
@awcrosby we have these markers `@pytest.mark.min_hub_version("4.6dev")
` and `@pytest.mark.max_hub_version("...")`. So we can create specific tests for those versions and tag them and they will be run or skipped accordingly

Reply:
That is cool we can limit tests to be run on different hub versions if needed. But my question was...

Say we merge a server-side change, and it gets deployed to c.stage.rh.c. We want to test if galaxy_ng 4.6.2 (and 4.5.z, 4.4.z) can continue to sync successfully against c.stage.rh.c.  

@chr-stian can we have Jenkins install 4.6.2 (and 4.5.z, 4.4.z), and then run this test? (with our without backporting this test into those branches)

Reply:
ah yes, we can. I defined the Jenkins job to not accept parameters because I thought that was what we needed but I can modify it to accept two branches: local hub branch and tests branch (if different). It's pretty straight-forward but it doesn't belong to this repo.

Reply:
ok cool... and if we do that, can the test run from galaxy_ng `master` hitting galaxy_ng `4.6.2`... or does the test need to run from galaxy_ng `stable-4.6` hitting galaxy_ng `4.6.2`?

either way sounds like this can be done after this PR

Reply:
right now tests are taken from the same branch as galaxy_ng has been started from. But I'll separate them so that we can have freedom to pick branches for hub and tests separately